internal/driver/driver.go

```diff
@@ -148,8 +159,71 @@ func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModels.A
 	// Make sure the paths of existing devices are up-to-date.
 	go d.RefreshExistingDevicePaths()
 
+	http.HandleFunc("/rtspauth", RtspCredentialsHandler)
+	go d.StartRtspCredentialServer()
 	return nil
 }
+func (d *Driver) StartRtspCredentialServer() {
+	d.lc.Infof("starting rtsp server\n")
+	err := http.ListenAndServe(":8000", nil) //TODO: get port from config
+	if err == http.ErrServerClosed {
+		d.lc.Errorf("server closed\n")
+	} else if err != nil {
+		d.lc.Errorf("error starting server: %s\n", err)
+	}
+}
+func RtspCredentialsHandler(w http.ResponseWriter, r *http.Request) {
+	//TODO: check for empty user and password
+	// from post
+	// {
+	// 	"user": "",
+	// 	"password": "",
+	//   }
+	//from post
+	// {
+	// 	"ip": "ip",
+	// 	"user": "user",
+	// 	"password": "password",
+	// 	"path": "path",
+	// 	"protocol": "rtsp|rtmp|hls|webrtc",
+	// 	"id": "id",
+	// 	"action": "read|publish",
+	// 	"query": "query"
+	//   }
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		driver.lc.Warnf("could not read body: %s\n", err)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	log.Println(string(body))
+	var rtspRequest RtspRequest
+	err = json.Unmarshal(body, &rtspRequest)
+	if err != nil {
+		driver.lc.Warnf("could not unmarshal read body: %s\n", err)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	fmt.Printf("user read body: %s\n", rtspRequest.User)
+	if rtspRequest.Action == "publish" {
+		driver.lc.Infof("allowing publish at path %s", rtspRequest.Path)
+		w.WriteHeader(http.StatusOK)
+	}
+	credential, edgexErr := driver.tryGetCredentials("rtsp_" + rtspRequest.User)
```

Comment:
i think the secret name should probably be a hard-coded constant. If we wanted to support more than one credential, it would probably be on a per-camera basis, so like based on the `rtspRequest.Path`. For now we can stick with one, and we can add the more advanced option if we see fit.

Reply:
i suggest maybe we go with a name like `rtspauth_default` or just `rtspauth` as the hard coded one, and then in the future we can expand it to support `rtspauth_<pathname>`, etc.